python/ml4ir/applications/classification/tests/test_tfrecord_reader.py

```diff
@@ -0,0 +1,129 @@
+from ml4ir.base.config.keys import TFRecordTypeKey
+from ml4ir.base.io.local_io import LocalIO
+from ml4ir.base.features.preprocessing import PreprocessingMap
+
+DATASET_PATH = "ml4ir/applications/classification/tests/data/tfrecord/train/file_0.tfrecord"
+FEATURE_CONFIG_PATH = "ml4ir/applications/classification/tests/data/configs/feature_config.yaml"
+
+
+class ExampleParserTest(unittest.TestCase):
+    """
+    Test class for ml4ir.base.data.tfrecord_reader.ExampleParser
+    """
+
+    def setUp(self):
+        file_io = LocalIO()
+        logger = logging.getLogger()
+
+        self.dataset = tf.data.TFRecordDataset(DATASET_PATH)
+        self.proto = next(iter(self.dataset))
+        self.feature_config = FeatureConfig.get_instance(
+            tfrecord_type=TFRecordTypeKey.EXAMPLE,
+            feature_config_dict=file_io.read_yaml(FEATURE_CONFIG_PATH),
+            logger=logger
+        )
+        self.parser = ExampleParser(feature_config=self.feature_config,
+                                    preprocessing_map=PreprocessingMap(),
+                                    required_fields_only=False)
+
+    def test_features_spec(self):
+        """
+        Test the feature specification constructed and used to parse the Example proto
+        """
+        features_spec = self.parser.features_spec
+
+        assert isinstance(features_spec, dict)
+
+        # Check if the feature specification matches with the feature_config
+        assert len(set(self.feature_config.get_all_features("name"))
+                   ) == len(features_spec)
+        for feature in self.feature_config.get_all_features("name"):
+            assert feature in features_spec
+
+    def test_extract_features_from_proto(self):
+        """
+        Test extraction of features from serialized proto
+        """
+        features = self.parser.extract_features_from_proto(
+            self.proto)
+
+        for feature, feature_tensor in features.items():
+            # Test that no feature is a sparse tensor
+            assert not isinstance(feature_tensor, tf.sparse.SparseTensor)
+
+            # Test that each extracted feature is a scalar
+            assert feature_tensor.shape == ()
+
+        # Assert that there is no mask feature
+        assert "mask" not in features
+
+    def test_get_default_tensor(self):
+        """
+        Test the default tensor used for missing features
+        """
+        default_tensor = self.parser.get_default_tensor(
+            self.feature_config.get_feature("query_text"))
+        assert default_tensor.shape == ()
+
+        default_tensor = self.parser.get_default_tensor(
+            self.feature_config.get_feature("user_context"))
+        assert default_tensor.shape == ()
+
+    def test_get_feature(self):
+        """
+        Test fetching feature tensor from extracted feature dictionary
+        """
+        feature_tensor = self.parser.get_feature(self.feature_config.get_feature("query_text"),
```

Comment:
It doesn't matter really. I just gave some random dimensions. The goal was to check if the dimensions will be expanded or not.

Reply:
I think it will make more sense in the follow up PR. I will set them to more sensible values. In this PR, think of this as also showing that the values don't really matter here.